### PR TITLE
#6Task. pa.artem

### PR DESCRIPTION
**Pull request by pa-artem**
https://travis-ci.org/github/AisBrestEDU/js-assignments/builds/713000210

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -30,7 +30,14 @@
  *
  */
 function getFizzBuzz(num) {
-    throw new Error('Not implemented');
+    if (num % 15 === 0)
+        return 'FizzBuzz';
+    else if (num % 3 === 0)
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: "else" is redundant after "return"

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -226,7 +258,12 @@ function reverseString(str) {
  *   34143 => 34143
  */
 function reverseInteger(num) {
-    throw new Error('Not implemented');
+    let result = 0;
+    for (let i = num.toString().length - 1; i >= 0; i--) {
+        result += (num % 10) * 10**i;
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: Indentation is missing

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -332,7 +399,35 @@ function isBracketsBalanced(str) {
...
+    function round(x) {
+        return x % 1 <= 0.5 ? Math.floor(x) : Math.ceil(x);
+    }
+    const s = 1000, min = 60 * s, h = 60 * min;
+    const d = 24 * h, m = 30 * d, y = 365 * d;
+
+    let interval = endDate - startDate;
+    if (interval <= 45 * s) {
+        return 'a few seconds ago';
+    } else if (interval <= 90 * s) {
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: "else" is redundant after "return"

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -431,7 +556,27 @@ function getMatrixProduct(m1, m2) {
...
-    throw new Error('Not implemented');
+    function win(c1, c2, c3) {
+        return (c1 === '0' || c1 === 'X') && c1 === c2 && c2 === c3;
+    }
+    
+    const dxs = [1, -1, 1, -1], dys = [1, 1, -1, -1];
+    const xs = [0, 2, 0, 2], ys = [0, 0, 2, 2];
+    for (let i = 0; i < 4; i++) {
+        let x = xs[i], y = ys[i], dx = dxs[i], dy = dys[i];
+        let isWin = win(position[x][y], position[x + dx][y + dy], position[x + 2 * dx][y + 2 * dy])
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: A simpler option would be to use the "if" statement without an additional variable

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -270,7 +313,20 @@ function isCreditCardNumber(ccn) {
  *   165536 (1+6+5+5+3+6 = 26,  2+6 = 8) => 8
  */
 function getDigitalRoot(num) {
-    throw new Error('Not implemented');
+    function sum(num) {
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: For this task you can use a simpler solution with one loop.

**Review by KostyaKov — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**